packages/catalog-platformstate-writer/src/utils.ts

```diff
@@ -100,3 +113,223 @@ export const readAllItems = async (
   const read: ScanCommandOutput = await dynamoDBClient.send(commandQuery);
   return read;
 };
+*/
+
+export const descriptorStateToClientState = (
+  state: DescriptorState
+): ItemState =>
+  state === descriptorState.published || state === descriptorState.deprecated
+    ? itemState.active
+    : itemState.inactive;
+
+export const updateDescriptorStateInPlatformStatesEntry = async (
+  dynamoDBClient: DynamoDBClient,
+  primaryKey: PlatformStatesEServiceDescriptorPK,
+  state: ItemState,
+  version: number
+): Promise<void> => {
+  const input: UpdateItemInput = {
+    ConditionExpression: "attribute_exists(PK)",
+    Key: {
+      PK: {
+        S: primaryKey,
+      },
+    },
+    ExpressionAttributeValues: {
+      ":newState": {
+        S: state,
+      },
+      ":newVersion": {
+        N: version.toString(),
+      },
+      ":newUpdateAt": {
+        S: new Date().toISOString(),
+      },
+    },
+    ExpressionAttributeNames: {
+      "#state": "state",
+    },
+    UpdateExpression:
+      "SET #state = :newState, version = :newVersion, updatedAt = :newUpdateAt",
+    TableName: config.tokenGenerationReadModelTableNamePlatform,
+    ReturnValues: "ALL_NEW",
```

Comment:
since updated value is not used, we can set this to
```suggestion
    ReturnValues: "NONE",
```

Reply:
Updated